src/core/trulens/core/utils/python.py

```diff
@@ -190,6 +191,9 @@ def safe_getattr(obj: Any, k: str, get_prop: bool = True) -> Any:
         if not get_prop:
             raise ValueError(f"{k} is a property")
 
+        if v.fget is None:
```

Comment:
this change doesn't have much to do with the type stuff, but I kept it so I don't have to split this into a separate PR, but basically this is just going to fail fast before L198.